src/org/jetbrains/java/decompiler/main/rels/MethodProcessorRunnable.java

```diff
@@ -219,6 +221,10 @@ public static RootStatement codeToJava(StructClass cl, StructMethod mt, MethodDe
 
       // Merge loop
       while (true) {
+        if (!root.hasLoops()) {
```

Comment:
why break out? wouldn't it be better to put the if around the `while (true)`?

Reply:
Hm. Good point. I thought it'd be cleaner, but I forgot about the other thing at the bottom